### PR TITLE
Integrate 8/20 Nightly RN Build (Extract RNTester)

### PR DESCRIPTION
**Pull request by NickGerleman**
This change integrates the 8/20 nightly react-native build (from the 8/19 build). The huge change here is that RNTester has been extracted into its own package. We do the same here, and adapt for a future in which this package is published, along with setting a pattern on how we can consume/patch Facebook monorepo packages.

`ReactCopies/RNTester` is moved to `@react-native/tester` to mirror the upstream package name. We use override tooling like before to ensure our copy is up to date.

We separate out patched versions of RNTester into `@react-native-windows/tester` and `react-native-win32-tester`. The build process of these packages involves a similar copy + patch process as before. These packages participate in override validation and their changes should automatically be picked up during integration.

Some limitations:
- We don't flow check these packages right now. We will want to do that if we start seeing product code in separate packages
- Playground's custom resolver doesn't know how to deal with these yet, so we need to do the traditional build for RNTester changes
- Integration tooling is about to get confused by having "overrides.json" files in packages that aren't RN forks. I will need to update integration logic, but want to in a separate PR since this is already quite large.

Core also did some gross bits with cross-package imports that we needed to patch around. Upstreaming these changes is tracked by #6210 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6211)

### CODE REVIEW

**Comment by NickGerleman** [comment]
A note for reviewers, any new file added from RNTester is just changing the relative import to something relative to react-native instead of the RNTester location in core's repo. I.e. nothing too interesting.

**Comment by NickGerleman** [comment]
Need to tweak the build/eslint configs a bit to not try to lint TypeScript sources that are copied to targets that we don't have type info for.

File: packages/react-native-win32/metro.config.js [hunk 1]
```diff
@@ -4,13 +4,24 @@
 const fs = require('fs');
 const path = require('path');
 
+const rnwTesterPath = fs.realpathSync(
+  path.dirname(require.resolve('react-native-win32-tester/package.json')),
```
Comment (dannyvv) [inline packages/react-native-win32/metro.config.js:8]: No need to change, just food for thought: I've seen this pattern in a few places, would it make sense to have a shared metro config that takes care of the tester and rnw?

**Comment by dannyvv** [comment]
lgtm, Just funny to see `win32` vs `windows` if we want to go short, should we just use `win` as I associate win32 generally with the oldest windows apis and MFC UI stack etc... like it is now macOS, not osx or os9 :)

**Review by dannyvv — APPROVED** [review]

File: packages/react-native-win32/metro.config.js (hunk 1 above)
Comment (NickGerleman) [reply packages/react-native-win32/metro.config.js:8]: There are some different enough semantics between configs that I think duplication is probably okay for now. We might want to start sharing more if we want to extract the resolver used by Playground though for use in other packages.

**Comment by NickGerleman** [comment]
Going to merge this one a bit aggressively since there will be about 50 bajillion merge conflicts™ otherwise.

**Comment by NickGerleman** [comment]
@msftbot merge in 1 minute

**Pull request merged**